### PR TITLE
[PRODDEV-150] Update filter breakpoints according to theme.

### PR DESCRIPTION
**Pull request by manachynskyi**
Link to the task: https://openy.atlassian.net/browse/PRODDEV-150

Steps for review:
- [ ] Login in as admin
- [ ] visit page where AF4 paragraph present (or create new)
- [ ] in new tab open theme settings page /admin/appearance to change theme for testing
- [ ] open AF setting form in new tab /admin/openy/settings/activity-finder
For dirrerent themes filters sibebar should be hidden on different resolutions (according to bootstrap version in AF settings form):
- [ ] Carnation. Should be set Bootstrap version - 4 https://i.imgur.com/ClrxSJd.png:
` < 1060` -filters' sidebar(left) hidden and displayed above results https://i.imgur.com/WXI6LYE.png
`>= 1060 `- filters' sidebar(left) displayed and top filters hidden https://i.imgur.com/PUaZDCz.png
- [ ] Lily, Rose. Should be set Bootstrap version - 3 https://i.imgur.com/sNnmBvm.png:
`< 992` - filters' sidebar(left) hidden and displayed above results https://i.imgur.com/pbGDRfc.png
`>= 992` - filters' sidebar(left)  displayed and top filters hidden https://i.imgur.com/TFnTAZv.png

### CODE REVIEW

File: src/Plugin/Block/ActivityFinder4Block.php [hunk 1]
```diff
@@ -146,6 +167,12 @@ public function build() {
 
     $sort_options = $backend->getSortOptions();
 
+    $bs_version = 4;
+    $theme_name = $this->themeManager->getActiveTheme()->getName();
```
Comment (duozersk) [inline src/Plugin/Block/ActivityFinder4Block.php]: There might be sites using the custom theme based on the rose / lily / carnation. Is there a way to check for the "base theme" here if this is the case?
And then there also might be sites using completely custom theme...

File: openy_af4_vue_app/src/components/ResultsBar.vue [hunk 2]
```diff
@@ -86,7 +86,7 @@ export default {
 </script>
 
 <style lang="scss">
-.results-bar-component {
+.results-bar-component .container {
```
Comment (duozersk) [inline openy_af4_vue_app/src/components/ResultsBar.vue]: Why do we need this change? If .container will have margins - then the background won't fill the whole width of the component.

File: src/Plugin/Block/ActivityFinder4Block.php [hunk 3]
```diff
@@ -175,6 +202,7 @@ public function build() {
         'mobile' => $image_mobile,
         'desktop' => $image_desktop,
       ],
+      '#bootstrap_version' => $bs_version,
```
Comment (duozersk) [inline src/Plugin/Block/ActivityFinder4Block.php]: "bootstrap_version" VS "bs_version" - could you please change it to use "bs_version", "bs-version" and "bsVersion"?

File: src/Plugin/Block/ActivityFinder4Block.php [hunk 4]
```diff
@@ -146,6 +167,12 @@ public function build() {
 
     $sort_options = $backend->getSortOptions();
 
+    $bs_version = 4;
```
Comment (duozersk) [inline src/Plugin/Block/ActivityFinder4Block.php]: BS 4 by default... sounds reasonable; but we initially developed it for the BS3. No need to change anything... just keep this in mind ;)

File: openy_af4_vue_app/src/App.vue [hunk 5]
```diff
@@ -537,6 +542,9 @@ export default {
     },
     showHomeBranchBlock() {
       return !this.hideHomeBranchBlock && this.homeBranchId
+    },
+    getResultsBarClasses() {
```
Comment (duozersk) [inline openy_af4_vue_app/src/App.vue]: I don't think "get..." is a good name for the computed property. Would advise to remove "get" from here and other computed property names. Just "resultsBarClasses".

File: openy_af4_vue_app/src/App.vue [hunk 6]
```diff
@@ -627,7 +635,9 @@ export default {
                 })
                 .filter(cartItem => cartItem.item)
             })
-            .catch(error => { error })
+            .catch(error => {
+              error
+            })
```
Comment (duozersk) [inline openy_af4_vue_app/src/App.vue:640]: Thank you for fixing these things ;)

File: openy_af4_vue_app/src/components/Results.vue [hunk 7]
```diff
@@ -8,7 +8,7 @@
         >
           <Loading />
         </div>
-        <div v-else class="col-12 col-xs-12 col-md-3 d-none d-md-block hidden-xs hidden-sm">
+        <div v-else :class="getFiltersClasses">
```
Comment (duozersk) [inline openy_af4_vue_app/src/components/Results.vue]: remove "get" => "filtersClasses"

File: openy_af4_vue_app/src/components/Results.vue [hunk 8]
```diff
@@ -18,7 +18,7 @@
         <div v-if="isLoadingData" class="col-12 col-xs-12 col-md-9">
           <Loading />
         </div>
-        <div v-else class="col-12 col-xs-12 col-md-9">
+        <div v-else :class="getResultsClasses">
```
Comment (duozersk) [inline openy_af4_vue_app/src/components/Results.vue]: remove "get"...

**Review by duozersk — CHANGES_REQUESTED** [review]
Great job! Please address the comments and we are good to go!

File: src/Plugin/Block/ActivityFinder4Block.php (hunk 1 above)
Comment (duozersk) [reply src/Plugin/Block/ActivityFinder4Block.php]: I even think about introducing the actual setting somewhere in admin form to have ability to set it there... out of scope for now.

File: openy_af4_vue_app/src/components/Results.vue (hunk 8 above)
Comment (manachynskyi) [reply openy_af4_vue_app/src/components/Results.vue]: Fixed.

File: openy_af4_vue_app/src/components/Results.vue (hunk 7 above)
Comment (manachynskyi) [reply openy_af4_vue_app/src/components/Results.vue]: Fixed.

File: openy_af4_vue_app/src/App.vue (hunk 5 above)
Comment (manachynskyi) [reply openy_af4_vue_app/src/App.vue]: Fixed.

File: src/Plugin/Block/ActivityFinder4Block.php (hunk 3 above)
Comment (manachynskyi) [reply src/Plugin/Block/ActivityFinder4Block.php]: Fixed.

File: openy_af4_vue_app/src/components/ResultsBar.vue (hunk 2 above)
Comment (manachynskyi) [reply openy_af4_vue_app/src/components/ResultsBar.vue]: We had a small style bug on tablet https://i.imgur.com/YJyHTEL.png and after this change it looks https://i.imgur.com/ryGSscV.png

File: src/Plugin/Block/ActivityFinder4Block.php (hunk 1 above)
Comment (manachynskyi) [reply src/Plugin/Block/ActivityFinder4Block.php]: For now, we get a base theme or if it missed an active theme and set related BS version. 
But the point with a separate field for it sounds cool.

File: openy_af4_vue_app/src/components/ResultsBar.vue (hunk 2 above)
Comment (duozersk) [reply openy_af4_vue_app/src/components/ResultsBar.vue]: Please remove this modification. It is not related to this issue.
And it should look like this on tablet:
image
the gray area should fill all the space from left to right.
The bug you referenced should probably be fixed in another way - using another page template for the landing page (it feels like there is already another .container higher in the HTML tree, and it shouldn't be there...)... or smth else.

**Review by duozersk — CHANGES_REQUESTED** [review]

File: openy_af4_vue_app/src/components/ResultsBar.vue (hunk 2 above)
Comment (manachynskyi) [reply openy_af4_vue_app/src/components/ResultsBar.vue]: Reverted.

**Review by duozersk — APPROVED** [review]

File: src/Plugin/Block/ActivityFinder4Block.php [hunk 9]
```diff
@@ -25,6 +26,12 @@ class ActivityFinder4Block extends BlockBase implements ContainerFactoryPluginIn
 
   use EntityBrowserFormTrait;
 
+  const THEME_BOOTSTRAP_VERSION = [
+    'openy_carnation' => 4,
```
Comment (podarok) [inline src/Plugin/Block/ActivityFinder4Block.php]: nice
But let's put this in config, please

**Review by podarok — CHANGES_REQUESTED** [review]

File: src/Plugin/Block/ActivityFinder4Block.php (hunk 9 above)
Comment (podarok) [reply src/Plugin/Block/ActivityFinder4Block.php]: @manachynskyi

Comment (manachynskyi) [reply src/Plugin/Block/ActivityFinder4Block.php]: Fixed.

**Review by duozersk — APPROVED** [review]

**Review by podarok — APPROVED** [review]

**Pull request merged**